bin/collator/src/parachain/service.rs

```diff
@@ -324,24 +329,47 @@ where
 
     // Frontier offchain DB task. Essential.
     // Maps emulated ethereum data to substrate native data.
-    task_manager.spawn_essential_handle().spawn(
-        "frontier-mapping-sync-worker",
-        Some("frontier"),
-        fc_mapping_sync::kv::MappingSyncWorker::new(
-            client.import_notification_stream(),
-            Duration::new(6, 0),
-            client.clone(),
-            backend.clone(),
-            storage_override.clone(),
-            frontier_backend.clone(),
-            3,
-            0,
-            fc_mapping_sync::SyncStrategy::Parachain,
-            sync_service.clone(),
-            pubsub_notification_sinks.clone(),
-        )
-        .for_each(|()| futures::future::ready(())),
-    );
+    match frontier_backend.as_ref() {
+        fc_db::Backend::KeyValue(ref b) => {
+            task_manager.spawn_essential_handle().spawn(
+                "frontier-mapping-sync-worker",
+                Some("frontier"),
+                fc_mapping_sync::kv::MappingSyncWorker::new(
+                    client.import_notification_stream(),
+                    Duration::new(6, 0),
+                    client.clone(),
+                    backend.clone(),
+                    storage_override.clone(),
+                    b.clone(),
+                    3,
+                    0,
+                    fc_mapping_sync::SyncStrategy::Parachain,
+                    sync_service.clone(),
+                    pubsub_notification_sinks.clone(),
+                )
+                .for_each(|()| futures::future::ready(())),
+            );
+        }
+        fc_db::Backend::Sql(ref b) => {
+            task_manager.spawn_essential_handle().spawn_blocking(
+                "frontier-mapping-sync-worker",
+                Some("frontier"),
+                fc_mapping_sync::sql::SyncWorker::run(
+                    client.clone(),
+                    backend.clone(),
+                    b.clone(),
+                    client.import_notification_stream(),
+                    fc_mapping_sync::sql::SyncWorkerConfig {
+                        read_notification_timeout: Duration::from_secs(10),
+                        check_indexed_blocks_interval: Duration::from_secs(60),
+                    },
```

Comment:
How did you pick the values?

Reply:
That's pretty much the default used by every impl in [frontier](https://github.com/search?q=fc_mapping_sync%3A%3Asql%3A%3ASyncWorkerConfig&type=code&p=3).

Reply:
Fair enough, I don't have a better suggestion 🙂